src/plugins/intel_gna/gna_graph_compiler.cpp

```diff
@@ -892,9 +918,19 @@ void GNAGraphCompiler::PoolingPrimitive(InferenceEngine::CNNLayerPtr layer) {
         getScaleFactor(layer, QuantizedDataType::output),
         ptr_inputs,
         ptr_outputs);
-    size_t num_data_bytes_out = InferenceEngine::details::product(begin(outputs->getDims()), end(outputs->getDims()))
-        * outputs->getPrecision().size();
+    size_t num_data_bytes_out = InferenceEngine::details::product(begin(outputs->getDims()), end(outputs->getDims()));
+
+    if (!is2DPooling) {
```

Comment:
Please add some explanation why this fix is enabled for all HW platforms.

Reply:
adding:
    // Need to reserve more memory otherwise the compiled model would not be
    // backward compatible with GNA 2.0
    // GNA 2.0 produces more outputs from 1D pooling than latter GNA generations (including GNA 3.0)
    // When the model is compiled for some newer GNA generation (than GNA 2.0)
    // but it does not use any specific new GNA features it should be correct to import and run using previous GNA HW